server/channels/web/saml.go

```diff
@@ -185,6 +196,36 @@ func completeSaml(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	c.App.AttachSessionCookies(c.AppContext, w, r)
 
+	desktopToken := relayProps["desktop_token"]
+	if desktopToken != "" {
+		expiryTime := time.Now().Add(-model.DesktopTokenTTL).Unix()
+		desktopTokenErr := c.App.AuthenticateClientDesktopToken(desktopToken, expiryTime, user)
+		if desktopTokenErr != nil {
+			handleError(desktopTokenErr)
+			return
+		}
+
+		serverToken, serverTokenErr := c.App.GenerateAndSaveServerDesktopToken(desktopToken, expiryTime)
+		if serverTokenErr != nil {
+			handleError(serverTokenErr)
+			return
+		}
+
+		queryString := map[string]string{
+			"server_token": *serverToken,
+		}
+		if val, ok := relayProps["redirect_to"]; ok {
+			queryString["redirect_to"] = val
+		}
+		if strings.HasPrefix(desktopToken, "dev-") {
+			queryString["isDesktopDev"] = "true"
+		}
+
+		redirectURL = utils.AppendQueryParamsToURL(c.GetSiteURLHeader()+"/login/desktop", queryString)
```

Comment:
Same comment as above, here we need to append the desktop client token as a query parameter too, so that the desktop can verify it against its local copy.